src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -51,5 +104,98 @@ public Assembly LoadFromPath(string fullPath)
                 return assembly;
             }
         }
+
+        private Assembly TryGetWellKnownAssembly(AssemblyLoadContext context, AssemblyName assemblyName)
+        {
+            if (!MSBuildLoadContext.WellKnownAssemblyNames.Contains(assemblyName.Name))
+            {
+                return null;
+            }
+
+            // Ensure we are attempting to load a matching version
+            // of the Microsoft.Build.* assembly.
+            assemblyName.Version = _currentAssemblyVersion;
+
+            var searchPaths = new[] { Assembly.GetExecutingAssembly().Location };
+            return TryResolveAssemblyFromPaths(context, assemblyName, searchPaths);
+        }
+
+        private Assembly TryResolveAssembly(AssemblyLoadContext context, AssemblyName assemblyName)
+        {
+            lock (_guard)
+            {
+                Assembly assembly = TryGetWellKnownAssembly(context, assemblyName);
+
+                if (assembly != null)
+                {
+                    return assembly;
+                }
+
+                if (_namesToAssemblies.TryGetValue(assemblyName.FullName, out assembly))
+                {
+                    return assembly;
+                }
+
+                return TryResolveAssemblyFromPaths(context, assemblyName, _dependencyPaths);
```

Comment:
This looks like mostly the same as the previous version—it just adds the possibility of cultures and custom search paths. Is that right?

Reply:
The same as what previous version? It should be the same as the v16.4 version.

Reply:
This was just a bad question. That's not what I meant, but I can't tell you what I did mean.